client/state/data-layer/wpcom/me/devices/index.js

```diff
@@ -0,0 +1,38 @@
+/**
+ * External dependencies
+ */
+import { translate } from 'i18n-calypso';
+import { keyBy } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import { http } from 'state/data-layer/wpcom-http/actions';
+import { dispatchRequest } from 'state/data-layer/wpcom-http/utils';
+import {
+	USER_DEVICES_REQUEST
+} from 'state/action-types';
+import { userDevicesAdd } from 'state/user-devices/actions';
+import { errorNotice } from 'state/notices/actions';
+
+export const requestUserDevices = function( { dispatch } ) {
+	dispatch( http( {
+		apiVersion: '1.1',
+		method: 'GET',
+		path: '/notifications/devices',
+	} ) );
+};
+
+export const handleSuccess = ( { dispatch }, action, next, devices ) => {
+	dispatch( userDevicesAdd( {
+		devices: keyBy( devices.map( ( { device_id, device_name } ) => ( { id: device_id, name: device_name } ) ), 'id' )
+	} ) );
+};
+
+export const handleError = ( { dispatch } ) => {
+	dispatch( errorNotice( translate( 'We couldn\'t load your devices, please try again.' ) ) );
+};
+
+export default {
+	[ USER_DEVICES_REQUEST ]: [ dispatchRequest( requestUserDevices, handleSuccess, handleError ) ],
```

Comment:
I'm not super into the idea. It would reduce boilerplate in a small number of cases for the simplest functions. 

 If it isn't an action creator then it can't even handle the cases where you need a single param in the http request